TexasHoldem.c

```diff
@@ -1,7 +1,456 @@
+//i am not too familiar with betting in this game so I will leave it to one of you guys to figure it out
+//i made this function for the bets, but I don't know how the blinds and different roles factor in
+void DefineBet(int playerNum, double playerbets[])
+{
+	//players are player1=0, player2=1, player3=2, and player4=3, going clockwise
+	int dealer = 0; //Start with first player=0 as dealer
+	int smallblind = (dealer+1)%4;
+	int bigblind = (dealer+2)%4;
+
+
+	//Define initial bets
+	switch(playerNum)
+	{
+		//one player 3 cpu
+		case 1:
+		printf("Player %d is the dealer, player %d is the small blind, and player %d is the big blind", dealer, smallblind, bigblind);
+		playerbets[smallblind] = 1;
+		playerbets[bigblind] = 2;
+		//cpus are players 1, 2, and 3
+		if ((dealer+3)%4 == 4)
+		{
+			printf("What is Player %d's bet? ", (dealer+3)%4);
+			scanf("%lf", playerbets[(dealer+3)%4]);
+		}
+		else
+		{
+			playerbets[(dealer+3)%4] = 2;
+		}
+		
+		
+		
+		break;
+		
+		//two players two cpu
+		//cpus are players 1 and 2
+		case 2:
+		printf("Player %d is the dealer, player %d is the small blind, and player %d is the big blind", dealer, (dealer+1)%4, (dealer+2)%4;
+		if ((dealer+3)%4 >= 3)
+		{
+			printf("What is Player %d's bet? ", (dealer+3)%4);
+			scanf("%lf", playerbets[(dealer+3)%4]);
+		}
+		else
+		{
+			playerbets[(dealer+3)%4] = 2;
+		}
+		
+		break;
+		
+		//three players one cpu
+		//cpu is player 1
+		case 3:
+		printf("Player %d is the dealer, player %d is the small blind, and player %d is the big blind", dealer, (dealer+1)%4, (dealer+2)%4;
+		if ((dealer+3)%4 >= 2)
+		{
+			printf("What is Player %d's bet? ", (dealer+3)%4);
+			scanf("%lf", playerbets[(dealer+3)%4]);
+		}
+		else
+		{
+			playerbets[(dealer+3)%4] = 2;
+		}
+		break;
+		
+		//four players no cpu
+		case 4:
+		printf("Player %d is the dealer, player %d is the small blind, and player %d is the big blind", dealer, (dealer+1)%4, (dealer+2)%4;
+		printf("What is Player %d's bet? ", (dealer+3)%4);
+		scanf("%lf", playerbets[(dealer+3)%4]);
+		break;
+	}
+}
+
+//this function essentially works as drawing the card for the player
+//playeridentifier selects what player the system is drawing for
+//draw is identifying what round is being drawn now
+//for example draw would be 1 if it is the first round of drawings
+void randomCard(int playeridentifier, int draw, int player1cards[][2], int player2cards[][2], int player3cards[][2], int player4cards[][2])
+{
+	int r, face, value;
+	r = rand();
+	face = (r% 4)+1;
+	value = (r% 13) +1;
+	switch(playeridentifier)
+	{
+		case 1:
+		player1cards[draw - 1][0] = face;
+		player1cards[draw - 1][1] = value;
+		break;
+		
+		case 2:
+		player2cards[draw - 1][0] = face;
+		player2cards[draw - 1][1] = value;
+		break;
+		
+		case 3:
+		player3cards[draw - 1][0] = face;
+		player3cards[draw - 1][1] = value;
+		break;
+		
+		case 4:
+		player4cards[draw - 1][0] = face;
+		player4cards[draw - 1][1] = value;
+		break;
+	}
+}
+
```

Comment:
We could do this the same way but just input 1 matrix since theres a cap of 4 players we dont really need a loop to iterate this function for us